packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -461,41 +464,53 @@ export class KubernetesClient {
   }
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  createV1Resource(client: CoreV1Api, manifest: any): Promise<any> {
+  createV1Resource(client: CoreV1Api, manifest: any, optionalNamespace?: string): Promise<any> {
     if (manifest.kind === 'Namespace') {
       return client.createNamespace(manifest);
     } else if (manifest.kind === 'Pod') {
-      return client.createNamespacedPod(manifest.metadata.namespace, manifest);
+      return client.createNamespacedPod(manifest.metadata.namespace || optionalNamespace, manifest);
```

Comment:
Shouldn't be the opposite? If the optionalNamespace is set it's because the `createResourcesFromFile` was called with it so it is explicitely set and has a higher priority.
optionalNamespace can be undefined, do you have to add `|| default`?

Reply:
@lstocchi I've switched the 3rd commit